### PR TITLE
Add queue handling

### PR DESCRIPTION
**Pull request by tomasz-rekawek**

### CODE REVIEW

File: src/js/tech/ChromecastTech.js [hunk 1]
```diff
@@ -230,52 +233,92 @@ ChromecastTech = {
...
+         if (customData) {
+            mediaInfo.customData = customData;
+         }
 
-      if (loadSource.credentials) {
-         request.credentials = loadSource.credentials;
-         request.credentialsType = loadSource.credentialsType;
-      }
+         this._ui.updateTitle(title);
+         this._ui.updateSubtitle(subtitle);
```
Comment (zemlanin) [inline src/js/tech/ChromecastTech.js:307]: Don't see these in `if (loadSource.sources)` branch

File: src/js/tech/ChromecastTech.js [hunk 2]
```diff
@@ -389,10 +432,15 @@ ChromecastTech = {
    ended: function() {
       var mediaSession = this._getMediaSession();
 
+      // Don't check for queues
+      // When handling a queue there are moments when mediaSession is null and current item has already finished
+      // and the new item is not started loading yet, which would end the session.
+      if (this._queue) {
+         return false;
```
Comment (zemlanin) [inline src/js/tech/ChromecastTech.js:439]: What if the whole queue has ended?

File: src/js/tech/ChromecastTech.js [hunk 3]
```diff
@@ -230,52 +233,92 @@ ChromecastTech = {
...
 
-      if (loadSource.credentials) {
-         request.credentials = loadSource.credentials;
-         request.credentialsType = loadSource.credentialsType;
-      }
+         this._ui.updateTitle(title);
+         this._ui.updateSubtitle(subtitle);
+         request = new chrome.cast.media.LoadRequest(mediaInfo);
+         request.autoplay = true;
+         request.currentTime = startTime === undefined ? loadSource.startTime : startTime;
```
Comment (zemlanin) [inline src/js/tech/ChromecastTech.js:310]: Or this

File: src/js/tech/ChromecastTech.js [hunk 4]
```diff
@@ -230,52 +233,92 @@ ChromecastTech = {
...
+            mediaInfoItem.duration = queueItem.duration;
+            mediaInfoItem.metadata.subtitle = queueItem.subtitle;
+            mediaInfoItem.streamType = this.videojsPlayer.liveTracker && this.videojsPlayer.liveTracker.isLive()
+               ? chrome.cast.media.StreamType.LIVE
+               : chrome.cast.media.StreamType.BUFFERED;
+            mediaInfoItem.tracks = [];
+            mediaInfoItem.activeTrackIds = [];
+
+            return new chrome.cast.media.QueueItem(mediaInfoItem);
+         });
```
Comment (zemlanin) [inline src/js/tech/ChromecastTech.js:262]: Can (most of) this callback be reused in both `if (loadSource.sources)` branches?

File: src/js/tech/ChromecastTech.js (hunk 3 above)
Comment (tomasz-rekawek) [reply src/js/tech/ChromecastTech.js]: https://github.com/standardbroadcast/videojs-chromecast/pull/2/commits/529213ce782c332924c811fc26e7c9d6bcd0805a

Comment (tomasz-rekawek) [reply src/js/tech/ChromecastTech.js]: https://github.com/standardbroadcast/videojs-chromecast/pull/2/commits/ced8dc2daf5445e36dba98608f2d71ab338c41aa

File: src/js/tech/ChromecastTech.js (hunk 4 above)
Comment (tomasz-rekawek) [reply src/js/tech/ChromecastTech.js]: I moved as much as possible of the code outside of conditional statements.

File: src/js/tech/ChromecastTech.js (hunk 2 above)
Comment (tomasz-rekawek) [reply src/js/tech/ChromecastTech.js:424]: I haven't found a reliable event. As in example when you are on the last chapter and you will use the chromechast control to switch to a different one the same events are fired as if the chapter would end...

Comment (zemlanin) [reply src/js/tech/ChromecastTech.js:424]: Better than nothing?..

Comment (tomasz-rekawek) [reply src/js/tech/ChromecastTech.js:424]: I think better UX is when the session stays hanging then when session is interrupted when you are attempting to watch previous episode

**Review by zemlanin — APPROVED** [review]

**Pull request merged**